controller.py

```diff
@@ -401,74 +431,139 @@ def add_with_metadata(self, collection_details):
+                sort_name=(entry.get("author") or Edition.UNKNOWN_AUTHOR),
+                roles=[Contributor.PRIMARY_AUTHOR_ROLE]
+            )
+            language = entry.get("dcterms_language")
+
+            presentation = PresentationCalculationPolicy(
+                choose_edition=False,
+                set_edition_metadata=False,
+                classify=False,
+                choose_summary=False,
+                calculate_quality=False,
+                choose_cover=False,
+                regenerate_opds_entries=False,
+            )
+            replace = ReplacementPolicy(presentation_calculation_policy=presentation)
+            metadata = Metadata(
+                data_source,
+                primary_identifier=IdentifierData(identifier.type, identifier.identifier),
+                title=title,
+                language=language,
+                contributors=[author],
+                links=links,
+            )
+
+            edition, ignore = metadata.edition(self._db)
+            metadata.apply(edition, collection, replace=replace)
 
             messages.append(message)
 
         title = "%s Catalog Item Additions for %s" % (collection.protocol, client.url)
-        url = cdn_url_for(
-            "add_with_metadata", collection_metadata_identifier=collection.name
-        )
+        url = self.collection_feed_url("add_with_metadata", collection)
         addition_feed = AcquisitionFeed(
             self._db, title, url, [], VerboseAnnotator,
             precomposed_entries=messages
         )
 
         return feed_response(addition_feed)
 
+    def metadata_requests_for(self, collection_details):
+        """Returns identifiers in the collection that could benefit from
+        distributor metadata on the circulation manager.
+        """
+        client = authenticated_client_from_request(self._db)
+        if isinstance(client, ProblemDetail):
+            return client
+
+        collection = collection_from_details(
+            self._db, client, collection_details
+        )
+
+        resolver = IdentifierResolutionCoverageProvider
+        unresolved_identifiers = collection.unresolved_catalog(
+            self._db, resolver.DATA_SOURCE_NAME, resolver.OPERATION
+        )
+
+        # Omit identifiers that currently have metadata pending for
+        # the IntegrationClientCoverageProvider.
+        data_source = DataSource.lookup(
+            self._db, collection.name, autocreate=True
+        )
+        is_awaiting_metadata = self._db.query(Identifier.id)\
+            .join(Identifier.coverage_records).filter(
+                CoverageRecord.data_source_id==data_source.id,
+                CoverageRecord.status==CoverageRecord.REGISTERED,
+                CoverageRecord.operation==IntegrationClientCoverageProvider.OPERATION,
+            ).subquery()
+
+        unresolved_identifiers = unresolved_identifiers.filter(
+            not_(Identifier.id.in_(is_awaiting_metadata))
+        )
```

Comment:
I'm concerned about the performance of this query when either the number of unresolved identifiers or the number of identifiers awaiting metadata is large. I think you could do an outer join against (coverage_records on data_source_id=data_source.id and status=REGISTERED and operation=OPERATION), and restrict the results to fields where that outer join doesn't turn up a record. Something like that should work.